### PR TITLE
Improve search-box behaviour, refs #5538 and #6111

### PR DESCRIPTION
**Pull request by jraddaoui**
- Allows the use of the 'tab' key to get out of the search-box
- Adds support to move over the result using the arrow keys
- Adds gray background to the selected result in the drop-down
- All the record is clickable (not just the link)
- The 'enter' key will redirect to the selected result or it will perform
  the query if no result is selected

### CODE REVIEW

File: js/dominion.js [hunk 1]
```diff
@@ -368,9 +370,51 @@
...
-    select: function (e) { },
+    next: function (e)
+      {
+        var $items = this.$menu.find('li');
+        var $active = this.$menu.find('li.active').first();
+
+        if ($active.length)
+        {
+          $active.removeClass('active');
+          $($items.get($items.index($active) + 1)).addClass('active');
```
Comment (sevein) [inline js/dominion.js]: `jQuery.get(index)` returns the DOM element but `jQuery.eq(index)` returns a jQuery object, so you don't have to call the constructor again.

File: js/dominion.js [hunk 2]
```diff
@@ -368,9 +370,51 @@
...
+        else
+        {
+          $items.first().addClass('active');
+        }
+      },
+
+    prev: function (e)
+      {
+        var $items = this.$menu.find('li');
+        var $active = this.$menu.find('li.active').first();
```
Comment (sevein) [inline js/dominion.js]: `'li.active:first'` may be faster.

File: js/dominion.js [hunk 3]
```diff
@@ -368,9 +370,51 @@
...
+
+        if ($active.length)
+        {
+          $active.removeClass('active');
+          $($items.get($items.index($active) - 1)).addClass('active');
```
Comment (sevein) [inline js/dominion.js]: You could use `jQuery.eq(index)` here too.

File: js/dominion.js [hunk 4]
```diff
@@ -368,9 +370,51 @@
...
+        }
+      },
+
+    select: function (e)
+      {
+        var $active = this.$menu.find('li.active').first();
+
+        if ($active.length)
+        {
+          $(location).attr('href',$active.find('a').attr('href'));
```
Comment (sevein) [inline js/dominion.js]: Whitespace after comma.

File: js/dominion.js [hunk 5]
```diff
@@ -482,8 +517,18 @@
...
+
+    mouseleave: function (e)
+      {
+        $(e.currentTarget).removeClass('active');
+      },
+
+    click: function (e)
+      {
+        e.preventDefault();
+        $(location).attr('href',$(e.currentTarget).find('a').attr('href'));
```
Comment (sevein) [inline js/dominion.js]: Whitespace after comma.

File: js/dominion.js [hunk 6]
```diff
@@ -368,9 +370,51 @@
...
+        }
+        else
+        {
+          $items.last().addClass('active');
+        }
+      },
+
+    select: function (e)
+      {
+        var $active = this.$menu.find('li.active').first();
```
Comment (sevein) [inline js/dominion.js]: `li.active:first`?

**Comment by jraddaoui** [comment]
Thanks @sevein !!

**Comment by jraddaoui** [comment]
Merged: https://github.com/artefactual/atom/commit/8b37612d65a5effabb8bf8e48ac706543b5cead3

**Pull request closed without merging**